Add simplification task and Duidelijke Taal dataset

simonevanbruggen opened the pull request:
- Added Duidelijke Taal dataset (see #1421). The create_duidelijke_taal.py script filters the original dataset, keeping the sentence pairs where there is a sufficient difference in complexity and the simplifications are evaluated as reasonably accurate. 
- Added simplification as a new task and included task configs and prompts (dataset is Dutch only, added English placeholder prompt for clarity for non-Dutch speakers).
- Added METEOR and SARI metrics. SARI requires the source text as input, so also included an extension to metrics/huggingface.py with a new SourceBasedMetric.

saattrupdan commented on docs/datasets/dutch.md:
```diff
@@ -1028,3 +1028,77 @@ You can evaluate this dataset directly as follows:
+
+## Simplification
+
+### Duidelijke Taal
```
Great to hear that you're willing to help out!

We don't need to evaluate all 432 models, thankfully. We've instead gone for the strategy of having a list of models that should be evaluated for all languages, which are continuously updated. This list can be found [in this pinned Github issue](https://github.com/EuroEval/EuroEval/issues/1186).

Here's a shell script that should evaluate all of those models as well as a few Dutch ones:

[run_duidelijke_taal_evaluations.sh](https://github.com/user-attachments/files/24310742/run_duidelijke_taal_evaluations.sh)

To run it, you need to have a `.env` file in your current working directory which sets `ANTHROPIC_API_KEY`, `GEMINI_API_KEY` and `OPENAI_API_KEY`. Since you're only evaluating a single dataset, this really won't cost a lost (just a few USD).

saattrupdan replied:
Note that this script of course depends on this PR having been merged in, so we should do that first 🙂 

We could assign the dataset as unofficial in this PR, and I could then change it to official when you've got the evaluation results (or you could create a tiny PR where you change it yourself even).

tvosch replied:
The local models sounds doable. We still need to setup multi-node vllm for the Llama 405B anyway (only 4 H100 per node) and get API keys for the three model providers anyway so indeed let's first get this PR into main :)
Thanks for the sharing the script!

simonevanbruggen commented on src/euroeval/tasks.py:
```diff
@@ -71,6 +72,16 @@
+    template_dict=SIMPL_TEMPLATES,
+    metrics=[m.meteor_metric, m.sari_metric, m.bert_score_metric],
+    default_num_few_shot_examples=3,
+    default_max_generated_tokens=128,
```
The longest target text is 43 words so should easily fit regardless of tokenizer. There are some longer texts in the original dataset, but they get filtered out.

simonevanbruggen commented on src/scripts/create_duidelijke_taal_nl.py:
```diff
@@ -0,0 +1,184 @@
+
+    # create 50%/50% train/val split
+    # original split is 50%/50% train/test, so total split becomes 25%/25%/50%
+    train_val = dataset_original["train"].train_test_split(test_size=0.5, seed=42)
```
Thanks for the suggestion, that's a really good idea. I had a look at the original dataset and ended up re-doing the "pre-splitted" dataset so it now has a similar complexity distribution. The filtering is now done before the additional train/val split and the split itself is also stratified on complexity. The distributions are still slightly different (with 4 bins, the difference between splits is 0 to 3 percent points) because of the prepared split, but it's much more balanced than before. 

Complexity of the source text seemed the best metric to balance the difficulty of the task. Fluency seemed a bit more subjective and the results of balancing on this made less sense to me. 

Re-doing the filtering/splitting slightly changed the resulting distribution, so this is updated in docs/datasets/dutch.md.

simonevanbruggen commented:
Thanks for the feedback and suggestions @saattrupdan! 

I addressed them in my last commits, and updated the dataset splitting/filtering (see comment above). 
`make check` is passing:
```
type annotations not comments............................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
debug statements (python)................................................Passed
ruff (legacy alias)......................................................Passed
ruff format..............................................................Passed
nbstripout...........................................(no files to check)Skipped
Pyrefly (type checking)..................................................Passed
markdownlint-cli2........................................................Passed
```

As for `CHECK_DATASET=duidelijke-taal-nl make test`, it passes all tests except `test_data_loading.py::TestAllDatasets::test_examples_in_official_datasets_are_not_too_long[duidelijke-taal-nl]` (since it can't seem to find the validation split) but it's passing when I run this with a locally saved version of the dataset. Could you see if there are any changes needed to fix this one?

saattrupdan commented:
> Thanks for the feedback and suggestions @saattrupdan!
> 
> I addressed them in my last commits, and updated the dataset splitting/filtering (see comment above). `make check` is passing:
> 
> ```
> type annotations not comments............................................Passed
> fix end of files.........................................................Passed
> trim trailing whitespace.................................................Passed
> debug statements (python)................................................Passed
> ruff (legacy alias)......................................................Passed
> ruff format..............................................................Passed
> nbstripout...........................................(no files to check)Skipped
> Pyrefly (type checking)..................................................Passed
> markdownlint-cli2........................................................Passed
> ```
> 
> As for `CHECK_DATASET=duidelijke-taal-nl make test`, it passes all tests except `test_data_loading.py::TestAllDatasets::test_examples_in_official_datasets_are_not_too_long[duidelijke-taal-nl]` (since it can't seem to find the validation split) but it's passing when I run this with a locally saved version of the dataset. Could you see if there are any changes needed to fix this one?

Great! I've updated the test now and merged main in with this branch - can you try again?

simonevanbruggen commented:
> > Thanks for the feedback and suggestions @saattrupdan!
> > I addressed them in my last commits, and updated the dataset splitting/filtering (see comment above). `make check` is passing:
> > ```
> > type annotations not comments............................................Passed
> > fix end of files.........................................................Passed
> > trim trailing whitespace.................................................Passed
> > debug statements (python)................................................Passed
> > ruff (legacy alias)......................................................Passed
> > ruff format..............................................................Passed
> > nbstripout...........................................(no files to check)Skipped
> > Pyrefly (type checking)..................................................Passed
> > markdownlint-cli2........................................................Passed
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > As for `CHECK_DATASET=duidelijke-taal-nl make test`, it passes all tests except `test_data_loading.py::TestAllDatasets::test_examples_in_official_datasets_are_not_too_long[duidelijke-taal-nl]` (since it can't seem to find the validation split) but it's passing when I run this with a locally saved version of the dataset. Could you see if there are any changes needed to fix this one?
> 
> Great! I've updated the test now and merged main in with this branch - can you try again?

Done, all passed!  🎉 
```
=================== 212 passed, 62 skipped in 275.88s (0:04:35) ===================
```

saattrupdan reviewed (approved):
Looks great, thanks a bunch for your contribution! Looking forward to the evaluations 🙂
